src/otx/algorithms/detection/adapters/mmdet/models/detectors/mean_teacher.py

```diff
@@ -182,22 +183,29 @@ def forward_train(
         pseudo_bboxes, pseudo_labels, pseudo_masks, pseudo_ratio = self.generate_pseudo_labels(
             teacher_outputs, device=current_device, img_meta=ul_img_metas, **kwargs
         )
+        non_empty = [bool(len(i)) for i in pseudo_labels] if self.filter_empty_annotations else [True]
+        get_unlabeled_loss = pseudo_ratio >= self.min_pseudo_label_ratio and any(non_empty)
+
+        if dist.is_initialized():
+            reduced_get_unlabeled_loss = torch.tensor(int(get_unlabeled_loss)).to(current_device)
+            dist.all_reduce(reduced_get_unlabeled_loss)
+            if dont_have_to_train := not get_unlabeled_loss and reduced_get_unlabeled_loss > 0:
+                get_unlabeled_loss = True
+                non_empty[0] = True
+
         if self.filter_empty_annotations:
```

Comment:
Please, put this block inside "if get_unalbeled_loss" statement, since we don't need to compute this when pseudo_ratio < self.min_pseudo_label_ratio. I know it wasn't written by you, but it just catches my eye.

Reply:
Thanks for comment. But it seems that if this statement is moved into `if get_unlabeled_loss`, how `self._visual_online()` works can be different, when `get_unlabeled_loss` is `True` and `self.visualize` is `False`. Is it ok?